### PR TITLE
Added list of open issues for each repository

### PR DESCRIPTION
**Pull request by AdityaA1998**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -25,6 +25,8 @@ dependencies {
     implementation  group: 'org.json', name: 'json', version: '20180130'
 
     testImplementation group: 'junit', name: 'junit', version: '4.12'
+
+    compile group: 'org.kohsuke', name: 'github-api', version: '1.93'
```
Comment (eugenepeh) [inline build.gradle]: compile is sort of a deprecated function, use  implementation instead and also group together with the other implementation.

File: src/main/java/reposense/dataobject/GithubIssue.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ */
+public class GithubIssue {
+
+    private final GithubIssueTitle title;
+    private final List<GithubIssueLabels> labels;
+    private final GithubIssueIndex index;
+    private final GithubIssueState state;
+
+    public GithubIssue(GithubIssueTitle title, List<GithubIssueLabels> labels,
+                       GithubIssueIndex index, GithubIssueState state) {
```
Comment (eugenepeh) [inline src/main/java/reposense/dataobject/GithubIssue.java]: wrapping indentation should be 8 spaces from previous line

File: src/main/java/reposense/dataobject/GithubIssueIndex.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Represents a github issue's index number
+ */
+public class GithubIssueIndex {
+
+    private int index;
+
+    /**
+     * Constructs a {@code GithubIssueIndex}.
```
Comment (eugenepeh) [inline src/main/java/reposense/dataobject/GithubIssueIndex.java]: redundant information, can be removed

File: src/main/java/reposense/gitmanagement/GitCredentials.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * Store github credentials
+ * Authenticate with github using the credentials
+ */
+public class GitCredentials {
+
+    private static final String USERNAME = "reposensetest";
+    private static final String TOKEN = "dd9a3a3d4aeebc57f45ef5903d2f891e5b6a8aaa";
+
+    private static GitHub github;
```
Comment (eugenepeh) [inline src/main/java/reposense/gitmanagement/GitCredentials.java]: if there's a constructor, I not sure why this has to be static

File: src/main/java/reposense/dataobject/GithubIssue.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Represents a github issue
+ */
+public class GithubIssue {
+
+    private final GithubIssueTitle title;
+    private final List<GithubIssueLabels> labels;
+    private final GithubIssueIndex index;
+    private final GithubIssueState state;
```
Comment (eugenepeh) [inline src/main/java/reposense/dataobject/GithubIssue.java]: If all these classes are all just going to be `String`, I not sure is it worth it to create a model for each of them.
This class on its own is also similar to `GHIssue` from `org.kohsuke.github`. Why not just use `GHIssue` instead?

**Comment by yong24s** [comment]
GitHub REST API does not require authentication for all endpoints.
See: https://api.github.com/repos/reposense/RepoSense/issues
We can perhaps inject dependency for user authentication when we detect a need to.

**Comment by eugenepeh** [comment]
Progress of this seems to have stopped, putting it on hold till you decide to continue.

**Comment by eugenepeh** [comment]
Closing this for now. Can reopen if you decide to continue this feature.

**Pull request closed without merging**